pkg/backends/cloudwatch/cloudwatch.go

```diff
@@ -21,34 +22,52 @@ const MAX_DIMENSIONS = 10
 // BackendName is the name of this backend.
 const BackendName = "cloudwatch"
 
+type QueueItem struct {
+	metrics  []*cloudwatch.MetricDatum
+	callback *gostatsd.SendCallback
+}
+
 // Client is an object that is used to send messages to AWS CloudWatch.
 type Client struct {
 	cloudwatch cloudwatchiface.CloudWatchAPI
 	namespace  string
 
+	batchDuration time.Duration
+
+	queue      []*QueueItem
+	queueMutex sync.RWMutex
+
+	timer      *time.Timer
+	timerMutex sync.Mutex
+
 	disabledSubtypes gostatsd.TimerSubtypes
 }
 
 // NewClientFromViper constructs a Cloudwatch backend.
 func NewClientFromViper(v *viper.Viper) (gostatsd.Backend, error) {
 	g := getSubViper(v, "cloudwatch")
 	g.SetDefault("namespace", "StatsD")
+	g.SetDefault("batchDuration", "50ms")
```

Comment:
`batchDuration` should be `batch-duration` for consistency, also needs documentation.

Reply:
Turns out there was no documentation for CW anyway /facepalm